### PR TITLE
python3Packages.zodb: fix compatibility with transaction 3.0

### PR DESCRIPTION
**Pull request by delroth**
###### Motivation for this change

zodb build broke with the python3Packages.transaction upgrade that
happened in f054f498400b285f0893727a2c039c802d771925. This cherry-picks a patch from upstream for compatibility.

@FRidh fyi

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/zodb/default.nix [hunk 1]
```diff
@@ -23,6 +24,14 @@ buildPythonPackage rec {
       sha256 = "20155942fa326e89ad8544225bafd74237af332ce9d7c7105a22318fe8269666";
     };
 
+    patches = [
+      # Compatibility with transaction v3.0
+      (fetchpatch {
+        url = "https://github.com/zopefoundation/ZODB/commit/0adcc6877f690186c97cc5da7e13788946d5e0df.patch";
+        sha256 = "122d6wkzgni704mifvgl1qzv353a6l6irlizhjkhmfqhpa05l3nh";
```
Comment (jonringer) [inline pkgs/development/python-modules/zodb/default.nix]: fetchpatch will do some normalization
```suggestion
        sha256 = "1zmbgm7r36nj5w7icpinp61fm81svh2wk213pzr3l0jxzr9i5qi4";
```

```
$ nix-build ./nixpkgs/ -A python3Packages.zodb
these derivations will be built:
  /nix/store/48dd5z5xgvybramnpn87369ck3sc98vy-0adcc6877f690186c97cc5da7e13788946d5e0df.patch.drv
  /nix/store/n4pia2rqck73304yffpid8kw8j608ybc-python3.7-ZODB-5.5.1.drv
building '/nix/store/48dd5z5xgvybramnpn87369ck3sc98vy-0adcc6877f690186c97cc5da7e13788946d5e0df.patch.drv'...

trying https://github.com/zopefoundation/ZODB/commit/0adcc6877f690186c97cc5da7e13788946d5e0df.patch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  7700    0  7700    0     0  45029      0 --:--:-- --:--:-- --:--:-- 45029
hash mismatch in fixed-output derivation '/nix/store/jzjpsk69j1nbwczjdpbra9gxcmx71zpx-0adcc6877f690186c97cc5da7e13788946d5e0df.patch':
  wanted: sha256:122d6wkzgni704mifvgl1qzv353a6l6irlizhjkhmfqhpa05l3nh
  got:    sha256:1zmbgm7r36nj5w7icpinp61fm81svh2wk213pzr3l0jxzr9i5qi4
cannot build derivation '/nix/store/n4pia2rqck73304yffpid8kw8j608ybc-python3.7-ZODB-5.5.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/n4pia2rqck73304yffpid8kw8j608ybc-python3.7-ZODB-5.5.1.drv' failed
```

Comment (delroth) [reply pkgs/development/python-modules/zodb/default.nix]: Ah, my bad, and using nix-prefetch-url hid the issue locally. Fixed.

**Comment by delroth** [comment]
@GrahamcOfBorg build python3Packages.zodb vulnix

**Review by jonringer — APPROVED** [review]
diff LGTM

failure related to python38Packages.zconfig
```
[2 built (1 failed), 0.0 MiB DL]
error: build of '/nix/store/xnr3j3i25gfil41kk7s0z5va7rzn5991-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/78443
1 package failed to build:
python38Packages.zodb

2 package built:
python37Packages.zodb vulnix
```

**Pull request merged**